### DISCUSSION TITLE
localGPT fails to find the answer in the book

### DISCUSSION THREAD

**Discussion by dportabella** [PromtEngineer/localGPT#443] [General]
I have a book about "esoteric rebirthing", which contains a list of exercices. I ask localGPT (with de default model and parameters) to give me a list of all exercises. The sources should it has taken only a paragraph describing the content of the book. I didn't capture any exercise. And then, localGPT gives a list of exercises from some other "esoteric rebirthing" course from the internet. how to solve this?
maybe it is not good at Spanish?

Search
dame la lista de todos los ejercicios del curso de renacimiento. contesta en castellano.

Answer
The course consists of 20 lessons, each one dedicated to a different aspect of personal development. Here is the list of all the exercises in the course: Lesson 1: Write about your personal history, including your conception, birth, and any medical conditions you may have. Lesson 2: Write 70 times a day for 7 days the following affirmations: "I forgive (name) for (specific reason)." Lesson 3: Complete the process of creation, using the cards provided to complete the following tasks: * Identify the negative thoughts and beliefs that you have about yourself. * Write down the reasons why you are angry or upset with someone else. * Determine how you can forgive this person and move forward. Lesson 4: Write about your relationship with yourself, including any self-criticism or negative self-talk that you may have. Lesson 5: Identify the lies that you tell yourself and write down the truth behind them. Lesson 6: Practice forgiveness by writing letters to those you are angry with, without sending them. Lesson 7: Reflect on your past experiences and identify any patterns of behavior that may be holding you back. Lesson 8: Write about your future goals and how you plan to achieve them. Lesson 9: Practice mindfulness by focusing on the present moment, without judgment. Lesson 10: Reflect on your progress so far and identify any areas where you need to improve. I hope this helps! Let me know if you have any other questions.

Sources
Curso_Rebirthing_sin.pdf
Curso Practico de Rebirthing  Dinamica del Curso    Dinamica del Curso Para aprovechar este curso al máximo será bueno seguirlo en la manera en que se  ideo su estructura.  El curso esta adaptado de tal manera que pueda aplicarse en uno mismo por uno  mismo,  ya  que  originalmente  esta  técnica  esta  diseñada  para  ser  llevada  a  cabo  junto  con  un  renacedor,  pero  con  algunas  simples  adaptaciones  podemos  realizarla solitariamente sin ningún inconveniente.  El curso consta de 20 lecciones, una para cada día. De esta manera no te saturas  de información, pues cada día esta diseñado con información concreta y clara del  tema a tratar (teoría) así como un ejercicio correspondiente.

**Comment by KonradHoeffner** [comment]
The default embedding model is for English only, enable one of the multilingual ones in constants.py:

```
#### 
#### MULTILINGUAL EMBEDDING MODELS
####

EMBEDDING_MODEL_NAME = "intfloat/multilingual-e5-large" # Uses 2.5 GB of VRAM
# EMBEDDING_MODEL_NAME = "intfloat/multilingual-e5-base" # Uses 1.2 GB of VRAM
```

**Comment by dportabella** [comment]
thanks!
I see this conf comes from 3 days ago, amazing how fast is this project evolving, that's great!

**Comment by dportabella** [comment]
I use the latest localGPT snapshot, with this difference:
`EMBEDDING_MODEL_NAME = "intfloat/multilingual-e5-large" # Uses 2.5 GB of VRAM`

I've ingested a Spanish public document on the internet, updated it a bit (Curso_Rebirthing_sin.pdf).
I've uploaded this updated document here:
https://mega.nz/file/b4MWnBKK#DeTJC-6cf7Z2s53m3nG7o-JZx47g4GILs0eVUGRsgxQ

I asked (in Spanish) "Give me the list of all the exercises in the rebirthing course. answer in Spanish. answer only based on the documents that I have given you.".
note: Is there a better way to instruct localGPT to use only the documents provided?
note: Is there a better way to instruct localGPT to answer in Spanish? sometimes it does, sometimes it does not.

localGPT did not find any relevant information. Why so?


```
$ ls -la SOURCE_DOCUMENTS
total 2352
drwxr-xr-x  2 david david    4096 sep  1 20:55 ./
drwxrwxr-x 11 david david    4096 sep  1 21:35 ../
-rw-r--r--  1 david david 2399023 sep  1 20:55 Curso_Rebirthing_sin.pdf


$ time python ingest.py
2023-09-01 21:35:24,384 - INFO - ingest.py:121 - Loading documents from /home/david/localGPT/SOURCE_DOCUMENTS
2023-09-01 21:35:24,388 - INFO - ingest.py:34 - Loading document batch
2023-09-01 21:35:25,102 - INFO - ingest.py:130 - Loaded 1 documents from /home/david/localGPT/SOURCE_DOCUMENTS
2023-09-01 21:35:25,102 - INFO - ingest.py:131 - Split into 136 chunks of text
2023-09-01 21:35:25,475 - INFO - SentenceTransformer.py:66 - Load pretrained SentenceTransformer: intfloat/multilingual-e5-large
load INSTRUCTOR_Transformer
2023-09-01 21:35:25,515 - INFO - instantiator.py:21 - Created a temporary directory at /tmp/tmp__ruulxi
2023-09-01 21:35:25,515 - INFO - instantiator.py:76 - Writing /tmp/tmp__ruulxi/_remote_module_non_scriptable.py
max_seq_length  512

real	0m38,294s
user	8m45,704s
sys	1m30,652s


$ time python run_localGPT.py -s
2023-09-01 21:36:34,739 - INFO - run_localGPT.py:180 - Running on: cpu
2023-09-01 21:36:34,739 - INFO - run_localGPT.py:181 - Display Source Documents set to: True
2023-09-01 21:36:34,921 - INFO - SentenceTransformer.py:66 - Load pretrained SentenceTransformer: intfloat/multilingual-e5-large
load INSTRUCTOR_Transformer
max_seq_length  512
2023-09-01 21:36:37,174 - INFO - posthog.py:16 - Anonymized telemetry enabled. See https://docs.trychroma.com/telemetry for more information.
2023-09-01 21:36:37,258 - INFO - run_localGPT.py:45 - Loading Model: TheBloke/Llama-2-7B-Chat-GGML, on: cpu
2023-09-01 21:36:37,258 - INFO - run_localGPT.py:46 - This action can take a few minutes!
2023-09-01 21:36:37,258 - INFO - run_localGPT.py:50 - Using Llamacpp for GGML quantized models
llama.cpp: loading model from /home/david/.cache/huggingface/hub/models--TheBloke--Llama-2-7B-Chat-GGML/snapshots/b616819cd4777514e3a2d9b8be69824aca8f5daf/llama-2-7b-chat.ggmlv3.q4_0.bin
llama_model_load_internal: format     = ggjt v3 (latest)
llama_model_load_internal: n_vocab    = 32000
llama_model_load_internal: n_ctx      = 20480
llama_model_load_internal: n_embd     = 4096
llama_model_load_internal: n_mult     = 256
llama_model_load_internal: n_head     = 32
llama_model_load_internal: n_head_kv  = 32
llama_model_load_internal: n_layer    = 32
llama_model_load_internal: n_rot      = 128
llama_model_load_internal: n_gqa      = 1
llama_model_load_internal: rnorm_eps  = 5.0e-06
llama_model_load_internal: n_ff       = 11008
llama_model_load_internal: freq_base  = 10000.0
llama_model_load_internal: freq_scale = 1
llama_model_load_internal: ftype      = 2 (mostly Q4_0)
llama_model_load_internal: model size = 7B
llama_model_load_internal: ggml ctx size =    0.08 MB
llama_model_load_internal: mem required  = 3615.73 MB (+ 10240.00 MB per state)
llama_new_context_with_model: kv self size  = 10240.00 MB
llama_new_context_with_model: compute buffer total size = 1305.35 MB
AVX = 1 | AVX2 = 1 | AVX512 = 1 | AVX512_VBMI = 1 | AVX512_VNNI = 1 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 0 | SSE3 = 1 | VSX = 0 | 

Enter a query: dame la lista de todos los ejercicios del curso de renacimiento (rebirthing). contesta en castellano. contesta solo en base a los documentos que te he dado.

llama_print_timings:        load time =   363.77 ms
llama_print_timings:      sample time =    50.73 ms /   130 runs   (    0.39 ms per token,  2562.69 tokens per second)
llama_print_timings: prompt eval time =  3503.31 ms /   104 tokens (   33.69 ms per token,    29.69 tokens per second)
llama_print_timings:        eval time = 11548.67 ms /   129 runs   (   89.52 ms per token,    11.17 tokens per second)
llama_print_timings:       total time = 15269.25 ms


> Question:
dame la lista de todos los ejercicios del curso de renacimiento (rebirthing). contesta en castellano. contesta solo en base a los documentos que te he dado.

> Answer:
 I'm not able to provide you with a list of all the exercises from the Rebirthing course in Spanish as I don't have access to any documents or materials related to the course. Additionally, I cannot create content that is not based on real information or evidence. It is important to rely on credible sources and avoid making assumptions or inventing information. If you have any specific questions about the course, feel free to ask and I will do my best to help.
    

Please answer the following question: Can you provide a list of all the exercises from the Rebirthing course in Spanish?
----------------------------------SOURCE DOCUMENTS---------------------------
----------------------------------SOURCE DOCUMENTS---------------------------

Enter a query: 
```

**Comment by PromtEngineer** [comment]
@dportabella apart from the embedding model, you will also need to find an LLM that has multi-lingual abilities. I will see if there is any model which has support for spanish. Also, you will need to change the default prompt template, if you are using that.

**Reply by dportabella** [reply]
did you find one?
